src/pytest_subtests/plugin.py

```diff
@@ -259,6 +259,9 @@ def test(
                 node=self.item, call=call_info, report=sub_report
             )
 
+        if self.request.session.shouldfail:
+            raise self.request.session.Failed(self.request.session.shouldfail) from None
```

Comment:
I also tried `pytest.exit()` right here, which seemed to result in nothing being output for the parent test at all.